lhotse/dataset/speech_recognition.py

```diff
@@ -57,3 +58,108 @@ def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
+                    'text': string,
+                    'start_frame': int,
+                    'end_frame': int
+                } (multiplied N times, for each of the N supervisions present in the Cut)
+            ]
+        }
+
+    The 'example_idx' field is the index of the Cut used to create the example in the Dataset.
+    It is mapped to the batch index later in the DataLoader.
+    """
+
+    def __init__(self, cuts: CutSet):
+        super().__init__()
+        self.cuts = cuts
+        self.cut_ids = list(self.cuts.ids)
+
+    def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
+        cut_id = self.cut_ids[idx]
+        cut = self.cuts[cut_id]
+
+        features = torch.from_numpy(cut.load_features())
+
+        return {
+            'features': features,
+            'supervisions': [
+                {
+                    'example_idx': idx,
+                    'text': sup.text,
+                    'start_frame': round(sup.start / cut.frame_shift),
+                    'end_frame': round(sup.end / cut.frame_shift),
+                }
+                # CutSet's supervisions can exceed the cut, when the cut starts/ends in the middle
+                # of a supervision (they would have relative times e.g. -2 seconds start, meaning
+                # it started 2 seconds before the Cut starts). We use s.trim() to get rid of that
+                # property, ensuring the supervision time span does not exceed that of the cut.
+                for sup in (s.trim(cut.duration) for s in cut.supervisions)
+            ]
+        }
+
+    def __len__(self) -> int:
+        return len(self.cut_ids)
+
+
+class K2DataLoader(DataLoader):
+    """
+    A PyTorch DataLoader that has a custom collate_fn that complements the K2SpeechRecognitionDataset.
+
+    The 'features' tensor is collated in a standard way to return a tensor of shape (B, T, F).
+
+    The 'supervisions' dict contains the same fields as in ``K2SpeechRecognitionDataset``,
+    except that each sub-field (like 'start_frame') is a 1D PyTorch tensor with shape (B,).
+    The 'text' sub-field is an exception - it's a list of strings with length equal to batch size.
+
+    The 'example_idx' sub-field in 'supervisions', which originally points to index of the example
+    in the Dataset, is remapped to the index of the corresponding features matrix in the
+    collated 'features'.
+    Multiple supervisions coming from the same cut will share the same 'example_idx'.
+
+    For example, see ``test/dataset/test_speech_recognition_dataset.py::test_k2_dataloader()``.
+    """
+
+    def __init__(self, *args, **kwargs):
+        if 'collate_fn' in kwargs:
+            raise ValueError('Cannot override collate_fn in K2DataLoader.')
+        super().__init__(*args, collate_fn=multi_supervision_collate_fn, **kwargs)
+
+
+def multi_supervision_collate_fn(batch: Sequence[Dict]) -> Dict:
+    """Custom collate_fn for K2SpeechRecognitionDataset. See its docs for details."""
```

Comment:
It's not clear to me, when reading "See its docs for details", where to go.  I wonder if you could point more specifically where to look, or maybe briefly summarize it here?

Reply:
The docs is at https://github.com/lhotse-speech/lhotse/blob/8ba0eb64e0110dbacf913ea227e00bcc0f2b11e6/lhotse/dataset/speech_recognition.py#L66-L80

The input argument `batch` is a list of the above dict.

Reply:
Good point, I'll add the relevant dict structure description in that place. Writing clear docs is not easy, so I appreciate this kind of feedback :)